ember-client/shared/src/main/scala/org/http4s/ember/client/internal/EmberWSClient.scala

```diff
@@ -47,8 +47,15 @@ private[client] object EmberWSClient {
         for {
           randomByteArray <- Resource.eval(random.nextBytes(16))
 
+          uriString = wsRequest.uri.renderString
+          newUriString = uriString match {
+            case x if x.startsWith("wss") => uriString.replaceFirst("wss", "https")
+            case x if x.startsWith("ws") => uriString.replaceFirst("ws", "https")
+            case _ => uriString
+          }
+
           httpWSRequest = Request[F]()
-            .withUri(wsRequest.uri)
+            .withUri(Uri.unsafeFromString(newUriString))
```

Comment:
Hmm, instead of going to/from the string representation, can we modify the schema directly on the Uri class?